src/control/server/faults.go

```diff
@@ -136,6 +136,14 @@ func FaultWrongSystem(reqName, sysName string) *fault.Fault {
 	)
 }
 
+func FaultIncompatibleComponents(self, other *build.VersionedComponent) *fault.Fault {
+	return serverFault(
+		code.ServerIncompatibleComponents,
+		fmt.Sprintf("components %s and %s are not compatible", self, other),
```

Comment:
I am probably missing something with golang, but it seems to be prettier like this ?
```suggestion
		fmt.Sprintf("components %s and %s are not compatible", self.String(), other.String()),
```